File: telegram-bot-bumblebee-core/src/main/java/com/github/bumblebee/command/youtube/entity/Subscription.java
package com.github.bumblebee.command.youtube.entity;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Entity
@Table(name = "BB_YOUTUBE_SUBSCRIPTIONS")
public class Subscription {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long subId;

    private String channelId;

    private Date updatedDate;

    @OneToMany(fetch = FetchType.EAGER, cascade = CascadeType.ALL,
            orphanRemoval = true, mappedBy = "subscription")
    private List<Chat> chats = new ArrayList<>();

    public String getChannelId() {
        return channelId;
    }

    public void setChannelId(String channelId) {
        this.channelId = channelId;
    }

    public Long getSubId() {
        return subId;
    }

    public void setSubId(Long subId) {
        this.subId = subId;
    }

    public List<Chat> getChats() {
        return chats;
    }

    public void setChats(List<Chat> chats) {
        this.chats = chats;
    }

    public Date getUpdatedDate() {
        return updatedDate;
    }

    public void setUpdatedDate(Date updatedDate) {
        this.updatedDate = updatedDate;
    }

    @Override
    public String toString() {
        return "Subscription{" +
                "subId=" + subId +
                ", channelId='" + channelId + '\'' +
                ", updatedDate=" + updatedDate +
                ", chats=" + chats +
                '}';
    }
}

File: telegram-bot-bumblebee-core/src/main/kotlin/com/github/bumblebee/command/youtube/entity/Subscription.kt
package com.github.bumblebee.command.youtube.entity

import java.util.*
import javax.persistence.*

@Entity
@Table(name = "BB_YOUTUBE_SUBSCRIPTIONS")
class Subscription {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var subId: Long? = null

    lateinit var channelId: String

    lateinit var updatedDate: Date

    @OneToMany(
        fetch = FetchType.EAGER,
        cascade = [(CascadeType.ALL)],
        orphanRemoval = true,
        mappedBy = "subscription"
    )
    var chats: MutableList<Chat> = ArrayList()

    override fun toString(): String =
        "Subscription(subId=$subId, channelId=$channelId, updatedDate=$updatedDate, chats=$chats)"
}
